Assignment.Tests/PingProcessTests.cs

```
@@ -1,152 +1,208 @@
+        {
+            Exception? inner = ex.Flatten().InnerException;
+            Assert.IsInstanceOfType<TaskCanceledException>(inner);
+            // Use exception.Flatten()
+        }
+    }
+
+    [TestMethod]
+    async public Task RunAsync_MultipleHostAddresses_True()
+    {
+        string[] hostNames = new string[] { "localhost", "localhost", "localhost", "localhost" };
+
+        int linesPerHost = PingOutputLikeExpression
+                           .Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries)
+                           .Length;
+        int expectedLineCount = linesPerHost * hostNames.Length;
+
+        PingResult result = await Sut.RunAsync(hostNames);
+
+        int actualLineCount = result.StdOutput?
+                              .Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries)
+                              .Length ?? 0;
+
+        Assert.AreEqual(expectedLineCount, actualLineCount);
+    }
+
+    [TestMethod]
+    async public Task RunLongRunningAsync_UsingTpl_Success()
+    {
+        ProcessStartInfo startInfo = new ProcessStartInfo("ping", "localhost");
+        StringBuilder outputBuilder = new();
+
+        PingResult result = await Sut.RunLongRunningAsync(
+            startInfo,
+            null,
+            null,
+            CancellationToken.None
+        );
+
+        Assert.AreEqual<int>(0, result.ExitCode);
+    }
+
+    [TestMethod]
+    public void StringBuilderAppendLine_InParallel_IsNotThreadSafe()
+    {
+        try
+        {
+            IEnumerable<int> numbers = Enumerable.Range(0, short.MaxValue);
+            System.Text.StringBuilder stringBuilder = new();
+            numbers.AsParallel().ForAll(item => stringBuilder.AppendLine(""));
+            int lineCount = stringBuilder.ToString().Split(Environment.NewLine).Length;
+            Assert.AreNotEqual(lineCount, numbers.Count() + 1);
+        }
+
+        catch (AggregateException)
+        {
+
```

You swallow the exception here when you catch it, I think it would be great to add some message or Assert.Fail message here to fulfill your try catch

---

Thanks for the feedback.

---

You're right, we did it quickly so the test wouldn't fail due to the exception. I just added an Assert to handle it properly now